pkg/scalers/kafka_scaler.go

```diff
@@ -289,46 +292,58 @@ func getKafkaClients(metadata kafkaMetadata) (sarama.Client, sarama.ClusterAdmin
 	return client, admin, nil
 }
 
-func (s *kafkaScaler) getPartitions() ([]int32, error) {
-	topicsMetadata, err := s.admin.DescribeTopics([]string{s.metadata.topic})
+func (s *kafkaScaler) getTopicPartitions() (map[string][]int32, error) {
+	var topicsToDescribe = make([]string, 0)
+
+	// when no topic is specified, query to cg group to fetch all subscribed topics
+	if s.metadata.topic == "" {
+		listCGOffsetResponse, err := s.admin.ListConsumerGroupOffsets(s.metadata.group, nil)
+		if err != nil {
+			return nil, fmt.Errorf("error listing cg offset: %s", err)
+		}
+		for topicName := range listCGOffsetResponse.Blocks {
+			topicsToDescribe = append(topicsToDescribe, topicName)
+		}
+	} else {
+		topicsToDescribe = []string{s.metadata.topic}
+	}
+
+	topicsMetadata, err := s.admin.DescribeTopics(topicsToDescribe)
 	if err != nil {
 		return nil, fmt.Errorf("error describing topics: %s", err)
 	}
-	if len(topicsMetadata) != 1 {
-		return nil, fmt.Errorf("expected only 1 topic metadata, got %d", len(topicsMetadata))
-	}
```

Comment:
Maybe we could keep this validation in case of `s.metadata.topic != ""`
something like:
```golang
         if s.metadata.topic != "" && len(topicsMetadata) != 1 {
		return nil, fmt.Errorf("expected only 1 topic metadata, got %d", len(topicsMetadata))
	}
```
WDYT?

Reply:
sure. let me add it

Reply:
added